File: Bua.CodeRev.CompilerService/Bua.CodeRev.CompilerService.Core/Models/CompilationError.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.CodeAnalysis;

namespace Bua.CodeRev.CompilerService.Core.Models
{
    public class CompilationError
    {
        public string ErrorCode { get; }
        public string Message { get; }
        public int StartChar { get; }
        public int EndChar { get; }
        public int StartLine { get; }
        public int EndLine { get; }

        public CompilationError(Diagnostic diagnostic)
        {
            var lineSpan = diagnostic.Location.GetLineSpan();

            ErrorCode = diagnostic.Id;
            Message = diagnostic.GetMessage();
            StartLine = lineSpan.StartLinePosition.Line;
            EndLine = lineSpan.EndLinePosition.Line;
            StartChar = lineSpan.StartLinePosition.Character;
            EndChar = lineSpan.EndLinePosition.Character;
        }
    }
}

File: CodeRev/WebClient/src/app/code-editor/models/compilationError.ts
import { CompilationErrorResponse } from './response/compilationError-response';

export class CompilationError {
    public errorCode: string;
    public message: string;
    public startChar: number;
    public endChar: number;
    public startLine: number;
    public endLine: number;

    constructor (resp: CompilationErrorResponse) {
        this.errorCode = resp.errorCode ?? '';
        this.message = resp.message ?? '';
        this.startChar = resp.startChar ?? 0;
        this.endChar = resp.endChar ?? 1;
        this.startLine = resp.startLine ?? 0;
        this.endLine = resp.endLine ?? 0;
    }
}